lib/xray/middleware.rb

```diff
@@ -59,7 +59,7 @@ def call(env)
           # Modifying the original response obj maintains compatibility with other middlewares
           if ActionDispatch::Response === response
             response.body = [body]
-            response.header['Content-Length'] = content_length
+            response.header['Content-Length'] = content_length unless response.try(:committed?)
```

Comment:
Oops, it is because the behavior of `try` has changed. Back in 3.2 it only checked for `nil`. Nowadays it also checks for `respond_to?`. So you're right, I'll need to add the `respond_to?` check to restore compatibility with 3.2. Good catch.

In the future can you open an issue instead of making a diff comment? It can be easy to lose track of a diff comment on a commit or a closed issue. In this case I'l take care of creating the issue. Thanks!